chainer/function_node.py

```diff
@@ -21,10 +21,13 @@
 import chainerx
 
 
-def _to_variable_with_chainerx_fallback_array(chainerx_array, fallback_array):
+def _to_variable_with_chainerx_fallback_array(
+        chainerx_device, chainerx_array, fallback_array):
     # chainerx_array can be None.
+    assert chainerx_array.device == chainerx_device.device
     var = variable.Variable(
         chainerx_array,
+        device=chainerx_device,
```

Comment:
How about using `_init_unchecked` here instead of `Variable.__init__`? Then we'd not need to introduce a new argument to the latter.

Reply:
Fixed as suggested.